backend/src/bork-protocol/utils/token-monitor/filter-tokens-by-interest.ts

```diff
@@ -0,0 +1,225 @@
+  }
+}
+
+/**
+ * Identifies new tokens from a list of tokens
+ * @param tokens List of tokens to check
+ * @param lastKnownTokens Set of token addresses that have been seen before
+ * @returns Array of new tokens and updated set of known tokens
+ */
+export function identifyNewTokens(
+  tokens: TokenProfile[],
+  lastKnownTokens: Set<string>,
+): { newTokens: TokenProfile[]; updatedKnownTokens: Set<string> } {
+  const updatedKnownTokens = new Set(lastKnownTokens);
+
+  const newTokens = tokens.filter(
+    (token) => !updatedKnownTokens.has(token.tokenAddress),
+  );
+
+  // Update our known tokens set
+  for (const token of tokens) {
+    updatedKnownTokens.add(token.tokenAddress);
+  }
+
+  return { newTokens, updatedKnownTokens };
+}
+
+/**
+ * Gets the most interesting tokens from a map
+ * @param interestingTokens Map of interesting tokens
+ * @param limit Maximum number of tokens to return
+ * @returns Array of interesting tokens sorted by score
+ */
+export function getTopInterestingTokens(
+  interestingTokens: Map<string, InterestingToken>,
+  limit = 5,
+): InterestingToken[] {
+  const tokens = Array.from(interestingTokens.values());
+
+  // Sort by score (highest first)
+  return tokens.sort((a, b) => b.score - a.score).slice(0, limit);
+}
+
+/**
+ * Checks for interesting tokens and initiates tweet searches
+ * @param interestingTokensGetter Function to get interesting tokens
+ * @param twitterService Twitter service for searching
+ * @param twitterConfigService Service to get Twitter configuration
+ * @param tweetQueueService Service to queue tweets for processing
+ * @param recentlySearchedTokens Set of tokens that were recently searched
+ * @returns The updated set of recently searched tokens
+ */
+export async function checkForInterestingTokens(
+  interestingTokensGetter: (limit: number) => InterestingToken[],
+  twitterService: TwitterService,
+  twitterConfigService: TwitterConfigService,
+  tweetQueueService: TweetQueueService,
+  recentlySearchedTokens: Set<string>,
+): Promise<Set<string>> {
+  elizaLogger.info('[TokenMonitor] Checking for interesting tokens');
+
+  try {
+    // Get interesting tokens
+    const interestingTokens = interestingTokensGetter(5);
+
+    if (interestingTokens.length === 0) {
+      elizaLogger.info('[TokenMonitor] No interesting tokens found');
+      return recentlySearchedTokens;
+    }
+
+    elizaLogger.info('[TokenMonitor] Found interesting tokens:', {
+      count: interestingTokens.length,
+      tokens: interestingTokens.map((t) => ({
+        address: t.tokenAddress,
+        reason: t.interestReason,
+        score: t.score,
+      })),
+    });
+
+    // Search for tweets about each token if not recently searched
+    let updatedRecentlySearchedTokens = new Set(recentlySearchedTokens);
+    for (const token of interestingTokens) {
+      if (!updatedRecentlySearchedTokens.has(token.tokenAddress)) {
+        updatedRecentlySearchedTokens = await searchTokenTweets(
```

Comment:
is that using a queue system? if not maybe add a todo, should be added to a queue IMO. Minor tho